node/lib/util/commit.js

```diff
@@ -2322,7 +2322,7 @@ exports.execSubmodulePrecommitHooks = co.wrap(function *(repo,
                     submoduleName
                 );
                 const status = yield Hook.execHook(subRepo, "pre-commit");
-                if (status === 1) {
+                if (status === "hook failed with no output") {
```

Comment:
We still want to stop if it fails with output (imagining that execHook were written correctly).

Reply:
ok. what would you recommend to do there? I am thinking for this function to return an object of shape: 
```
interface Result {
    code: number;
    status: string;
}
```
what do you think?

Reply:
I just pushed a change to address this. The execHook function now returns a boolean indicating 'isOk'. By 'ok', we mean no error, that is, either the hook doesn't exist or hook ran successfully without errors.